### ISSUE TITLE
Rework next_phase and previous_phase methods on Game

### ISSUE THREAD

**Issue by mayberryrfd** [mattdeboard/rs-pugbot#23]
I think it'd be better if all phase transition logic, including sending embeds (i.e. Discord messages) back to Discord at phase transition lines, were to be handled [here](https://github.com/mayberryrfd/rs-pugbot/blob/master/src/pugbot/models/game.rs#L196-L237).  Calling `game.next_phase()` should do all the heavy lifting required to move the game forward both for the bot itself as well as the users. 

This will make issues like #22, where we want to add a new phase, a lot more straightforward.

For example, I'd like to see code like [this](https://github.com/mayberryrfd/rs-pugbot/blob/master/src/pugbot/commands/add.rs#L31-L42) and [this](https://github.com/mayberryrfd/rs-pugbot/blob/master/src/pugbot/commands/mapvote.rs#L34) moved out of the commands and just be a part of the `next_phase` and `previous_phase` logic.

**Comment by mayberryrfd** [comment]
[here](https://github.com/mayberryrfd/rs-pugbot/blob/master/src/pugbot/models/game.rs#L122-L124) is another pretty gruesome example. Calls to `next_phase` are just scattered all over, ditto embed-creation logic. Let's get this into one spot.

**Comment by mattdeboard** [comment]
I'd like to restructure layout here so command handlers are much simpler
```rust
command!(some_command(ctx, msg) {
  let mut data = ctx.data.lock();
  let game = data.get_mut::<Game>().unwrap(); 

  if let Ok(_) = essential_logic_of_command() {
    game.next_phase();
  } else {
    handle_failure();
  }
})

fn essential_logic_of_command() {
  // ...
}

fn handle_failure() {
  // ...
}
```

This seems like the right design. At least an improvement over the current state of things.

**Comment by mattdeboard** [comment]
I made this flowchart to illustrate what I’m thinking here.

![7f97a6ba-187b-4497-b62d-7ad04dfcba49](https://user-images.githubusercontent.com/306571/50500488-4f271e00-0a17-11e9-8079-d47f35544009.png)

**Comment by mattdeboard** [comment]
Scope of this issue was fixed by #25, #29 and #30. Nice job, me!
